src/main/java/br/com/deveficiente/nossomercadolivreapi/compra/nova/EfetuaCompraController.java

```diff
@@ -0,0 +1,66 @@
+package br.com.deveficiente.nossomercadolivreapi.compra.nova;
+
+import br.com.deveficiente.nossomercadolivreapi.gatewaypagamento.GatewayPagamentoType;
+import br.com.deveficiente.nossomercadolivreapi.produto.NotificaDonoDoProdutoService;
+import br.com.deveficiente.nossomercadolivreapi.produto.Produto;
+import br.com.deveficiente.nossomercadolivreapi.produto.ProdutoRepository;
+import br.com.deveficiente.nossomercadolivreapi.usuario.Usuario;
+import br.com.deveficiente.nossomercadolivreapi.usuario.UsuarioRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.validation.Valid;
+
+@RestController
+public class EfetuaCompraController {
```

Comment:
Essa classe está com 13(se eu contei certo) pontos de carga cognitiva. Confesso que esse é um número que ainda estou buscando e considerando que ele é um controller 100% coeso, a carga cognitiva pode ser maior sim. Mas, mesmo neste cenário, onde você tem uma classe que realiza o fluxo de negócio mesmo, eu acho que a carga cognitiva máxima deveria ser 7 (feeling total). Quer tentar refatorar para sentir como fica?

Neste momento eu meço a carga cognitiva pelo acoplamento com classes específicas do próprio projeto + número de branches(if, loop).

Reply:
Eu estava implementando com uma classe para tratar o fluxo da compra "EfetuaCompraService" hehe o bendito Service, mas como na live nós estavámos falando justamente sobre esse assunto, que poderia deixar o fluxo no próprio controller por ele ser coeso e tals eu acabei deixando quieto.

Implementei uma classe "EfetuaCompraService" e joguei o fluxo pra lá.

Veja se meu entendimento está correto:

Devemos criar classes coesas mas que tenham "poucos" ponto de carga cognitiva, se a carga aumentar muito, devemos criar novas classes para dividir essa carga.

Outra coisa é deixar/forçar com que o código seja auto explicativo, nem que para isso uma classe/abstração seja criada.